src/RuleTicket.php

```diff
@@ -82,6 +82,46 @@ public function executeActions($output, $params, array $input = [])
                             $output['locations_id'] = $output['_locations_id_of_item'];
                         }
                         break;
+
+                    case 'regex_result':
+                        // Get each regex values
+                        $regex_values = array_map(
+                            fn ($regex_result) => RuleAction::getRegexResultById(
+                                $action->fields["value"],
+                                $regex_result
+                            ),
+                            $this->regex_results
+                        );
+
+                        // Get field
+                        $field = $action->fields["field"];
+
+                        // Handle each fields
+                        if ($field == "type") {
+                            foreach ($regex_values as $regex_value) {
+                                // Get defined ticket types
+                                // Note that these types are translated to the
+                                // current user language, which mean the rule
+                                // will only work for one language...
+                                // For example, if the regex value is "Request"
+                                // and the logged in user is french then "Request"
+                                // wont match "Demande" and the rule wont work
+                                // As far as I known, they are no workaround
+                                // for this and it should be considered a
+                                // limitation of the rule engine when working
+                                // with regex results
```

Comment:
I am not sure to understand (rules engine is still a black box for me). You want to assign the type that corresponds to a string in another ticket field (e.g its name) ?
In this case, it seems correct to have such limitation, but I am not sure that this feature will be really usable. Indeed, translations may vary along time, and user may also use a synonym. IMHO, it would be better to propose to administrator to use a rule like following one:
> If `name` matches `/^(Request|Demande|Requête|Besoin)\s+/`, then `assign` `2` to `type`.

With such a rule, administrator will be able to easilly add synonyms, translations, and even words with common typos.

Reply:
> I am not sure to understand. You want to assign the type that corresponds to a string in another ticket field (e.g its name) ?

Yes this is what was defined in the specification (which I wasn't involved in; and I do agree that your suggested use of the existing rules action seems better).
I imagine that the user will have the type contained in the category (Category: Incident - xxxxx) or some other field.

Reply:
I guess the customer would be able to understand that if we propose a safe alternative on hazardous feature request (due to a technical limitation), then the alternative solution should be used.

Reply:
Done, waiting for answer.

Reply:
Client agreed, I've removed this rule in [423fe18](https://github.com/glpi-project/glpi/pull/12301/commits/423fe18d3cd16bac95bb621148f1cc8d880e3cbc).